File: app/src/main/java/com/tronography/pixelweather/weather/Weather.java
package com.tronography.pixelweather.weather;

import com.tronography.pixelweather.model.CurrentWeatherModel;
import com.tronography.pixelweather.model.ForecastModel;
import com.tronography.pixelweather.model.WeatherReport;

import java.util.List;


public interface Weather {

    interface View {

        void showLoading(boolean show);

        void showError(String error);

        void showToast(String message);

        void showWeatherReport(WeatherReport weatherReport);
    }
}

File: app/src/main/java/com/tronography/pixelweather/weather/Weather.kt
package com.tronography.pixelweather.weather

import com.tronography.pixelweather.model.WeatherReport


interface Weather {

    interface View {

        fun showLoading(show: Boolean)

        fun showError(error: String?)

        fun showWeatherReport(weatherReport: WeatherReport)
    }
}